In [1]:
# Import Libraries to read and plot datasets
import pandas as pd
import numpy as np
import os
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.decomposition import PCA
from sklearn.preprocessing import scale
from collections import Counter
import regex

In [2]:
crash_data = pd.read_csv("hq1d-p-app52dopendataexport2025crashes.csv")


C:\Users\Evan\AppData\Local\Temp\ipykernel_19796\1470525132.py:1: DtypeWarning: Columns (0: 	NCIC Code, 1: 	City Is Active, 2: 	City Is Incorporated, 3: 	IsHighwayRelated, 4: 	IsTowAway, 5:  Road Condition 2, 6: MilepostUnitOfMeasure, 7: 	ReportingDistrictCode, 8: 	SketchDesc, 9: IsCountyRoad, 10: IsFreeway, 11: 	IsAdditonalObjectStruck, 12: NotificationDate, 13: 	NotificationTimeDescription, 14: 	HasDigitalMediaFiles, 15: 	EvidenceNumber, 16: 	IsLocationReferToNarrative, 17: 	IsAOIOneSameAsLocation) have mixed types. Specify dtype option on import or set low_memory=False.
  crash_data = pd.read_csv("hq1d-p-app52dopendataexport2025crashes.csv")


In [3]:
len(crash_data)

393510

In [4]:
crash_data.head()

,Collision Id,\tReport Number,\tReport Version,\tIs Preliminary,\tNCIC Code,Crash Date Time,\tCrash Time Description,\tBeat,City Id,\tCity Code,...,IsCountyRoad,IsFreeway,CHP555Version,\tIsAdditonalObjectStruck,NotificationDate,\tNotificationTimeDescription,\tHasDigitalMediaFiles,\tEvidenceNumber,\tIsLocationReferToNarrative,\tIsAOIOneSameAsLocation
0,4550265,9685-2025-00096,1,False,9685,1/18/2025 3:20:00 PM,1520,039,994.0,3300.0,...,False,False,4,NaN,1/18/2025 3:25:00 PM,1525,False,NaN,NaN,True
1,4550264,250110005,1,False,4116,1/10/2025 8:28:00 AM,828,A,1310.0,4116.0,...,NaN,NaN,4,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,4550263,9580-2025-00214,1,False,9580,1/23/2025 8:45:00 AM,845,171,535.0,1942.0,...,False,True,4,NaN,1/23/2025 8:47:00 AM,0847,False,NaN,NaN,True
3,4550262,9765-2025-00066,1,False,9765,1/15/2025 12:30:00 PM,1230,041,1701.0,5606.0,...,False,True,4,NaN,1/15/2025 12:35:00 PM,1235,True,EMBEDDED,NaN,True
4,4550261,9426-2025-00035,1,False,9426,1/20/2025 12:30:00 PM,1230,460,397.0,1500.0,...,True,False,4,NaN,1/20/2025 12:32:00 PM,1232,False,NaN,NaN,True


In [5]:
crash_data.columns = crash_data.columns.str.strip()

In [6]:
crash_parties_data = pd.read_csv("a2676918-a825-4b77-8e5c-6eadb38d6b1a.csv")

In [7]:
len(crash_parties_data)

767052

In [8]:
crash_parties_data

,_id,PartyId,CollisionId,PartyNumber,PartyType,IsAtFault,IsOnDutyEmergencyVehicle,IsHitAndRun,AirbagCode,AirbagDescription,...,Vehicle2TypeDesc,Vehicle2Year,Vehicle2Make,Vehicle2Model,Vehicle2Color,V2IsVehicleTowed,Lane,ThruLanes,TotalLanes,IsDREConducted
0,1,13900346,4541902,1,Driver,True,False,True,B,UNKNOWN,...,NaN,NaN,NaN,NaN,NaN,NaN,N/B,1.0,1.0,False
1,2,13900345,4541901,2,Driver,NaN,False,False,M,AIR BAG NOT DEPLOYED,...,NaN,NaN,NaN,NaN,NaN,NaN,2,2.0,5.0,False
2,3,13900344,4541901,1,Driver,True,False,False,L,AIR BAG DEPLOYED,...,NaN,NaN,NaN,NaN,NaN,NaN,2,2.0,5.0,False
3,4,13900343,4541900,2,Driver,NaN,False,False,M,AIR BAG NOT DEPLOYED,...,NaN,NaN,NaN,NaN,NaN,NaN,E/B,1.0,1.0,False
4,5,13900342,4541900,1,Driver,NaN,False,False,M,AIR BAG NOT DEPLOYED,...,NaN,NaN,NaN,NaN,NaN,NaN,N/B,1.0,1.0,False
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
767047,767048,14979609,5105674,1,Driver,True,False,NaN,M,AIR BAG NOT DEPLOYED,...,NaN,NaN,NaN,NaN,NaN,NaN,1,1.0,1.0,NaN
767048,767049,14979608,5105673,1,Driver,True,False,NaN,M,AIR BAG NOT DEPLOYED,...,NaN,NaN,NaN,NaN,NaN,NaN,1,1.0,1.0,NaN
767049,767050,14979607,5105672,2,Driver,NaN,False,NaN,M,AIR BAG NOT DEPLOYED,...,NaN,NaN,NaN,NaN,NaN,NaN,1,3.0,3.0,NaN
767050,767051,14979606,5105672,1,Driver,True,False,NaN,M,AIR BAG NOT DEPLOYED,...,NaN,NaN,NaN,NaN,NaN,NaN,2,3.0,3.0,NaN


In [9]:
crash_data = crash_data.drop(['Report Number','Report Version','Is Preliminary','NCIC Code','City Is Active',
       'City Is Incorporated', 'Collision Type Code','HasPhotographs', 'IsAttachmentsMailed', 'IsDeleted','JudicialDistrict', 'MotorVehicleInvolvedWithCode', 'MilepostDirection', 'MilepostDistance', 'MilepostMarker',
       'MilepostUnitOfMeasure', 'PreparedDate','ReportingDistrict',
       'ReportingDistrictCode', 'ReviewedDate', 'RoadwaySurfaceCode',
       'SecondaryDirection', 'SecondaryDistance', 'SecondaryRoad',
       'SecondaryUnitOfMeasure', 'SketchDesc', 'TrafficControlDeviceCode',
       'CreatedDate', 'ModifiedDate','CHP555Version', 'NotificationTimeDescription', 'HasDigitalMediaFiles', 'EvidenceNumber',
       'IsLocationReferToNarrative', 'DispatchNotified','Crash Time Description', 'Beat', 'City Id', 'City Code','Weather 2','Road Condition 2','Special Condition','Collision Type Other Desc','NotificationDate', 'IsAOIOneSameAsLocation',
        'IsCountyRoad'], axis = 1)

In [10]:
crash_data.columns

Index(['Collision Id', 'Crash Date Time', 'City Name', 'County Code',
       'Collision Type Description', 'Day Of Week', 'HitRun',
       'IsHighwayRelated', 'IsTowAway', 'MotorVehicleInvolvedWithDesc',
       'MotorVehicleInvolvedWithOtherDesc', 'NumberInjured', 'NumberKilled',
       'Weather 1', 'Road Condition 1', 'LightingCode', 'LightingDescription',
       'Latitude', 'Longitude', 'PedestrianActionCode', 'PedestrianActionDesc',
       'Primary Collision Factor Code', 'Primary Collision Factor Violation',
       'PrimaryCollisionFactorIsCited', 'PrimaryCollisionPartyNumber',
       'PrimaryRoad', 'IsFreeway', 'IsAdditonalObjectStruck'],
      dtype='str')

In [11]:
crash_data_roads = crash_data[['Collision Id','PrimaryRoad','IsFreeway','IsHighwayRelated']]

In [12]:
crash_data_roads.head()

,Collision Id,PrimaryRoad,IsFreeway,IsHighwayRelated
0,4550265,SR-79 N/B,False,True
1,4550264,N EL CAMINO REAL,NaN,False
2,4550263,SR-170 S/B FROM SHERMAN WAY,True,True
3,4550262,SR-126 E/B,True,True
4,4550261,SR-46 W/B,False,False


In [13]:
crash_data_roads['IsFreeway'] = crash_data_roads['IsFreeway'].fillna(crash_data_roads['IsHighwayRelated']).fillna(False)


#crash_data_roads = crash_data_roads.fillna(False, axis = 1) #fill in NA fields with False since it should only be True or False

In [14]:
crash_data_roads.count() # confirming that the fields that should be here

Collision Id        393510
PrimaryRoad         393510
IsFreeway           393510
IsHighwayRelated    393509
dtype: int64

In [15]:
freeway_crashes = crash_data_roads[crash_data_roads['IsFreeway'] == True]

In [16]:
local_crashes = crash_data_roads[crash_data_roads['IsFreeway'] == False]

In [17]:
local_crashes

,Collision Id,PrimaryRoad,IsFreeway,IsHighwayRelated
0,4550265,SR-79 N/B,False,True
1,4550264,N EL CAMINO REAL,False,False
4,4550261,SR-46 W/B,False,False
7,4550258,SR-74 (ORTEGA HWY) W/B,False,True
10,4550253,83RD ST,False,False
...,...,...,...,...
393504,5105492,PROSPECT ST,False,False
393505,5105490,MAIN ST,False,False
393506,5105489,LA VETA AVE,False,False
393507,5105488,CANNON ST,False,False


In [18]:
assert(len(local_crashes) + len(freeway_crashes)) == 393510 # assert that the divide between local and highway crashes add up to the original amount

In [19]:
local_crashes.to_csv("local_crashes.csv")

In [20]:
freeway_crashes.to_csv("freeway_crashes.csv")

In [21]:
crash_data_roads.to_csv("crash_data_roads.csv")

In [22]:
crash_data[crash_data['IsAdditonalObjectStruck'].isna() ==True]['IsAdditonalObjectStruck']

0         NaN
1         NaN
2         NaN
3         NaN
4         NaN
         ... 
393505    NaN
393506    NaN
393507    NaN
393508    NaN
393509    NaN
Name: IsAdditonalObjectStruck, Length: 370507, dtype: object

In [23]:
crash_data[crash_data['IsAdditonalObjectStruck'] ==True]['IsAdditonalObjectStruck']

19        True
46        True
56        True
58        True
62        True
          ... 
392787    True
392798    True
392806    True
392811    True
392815    True
Name: IsAdditonalObjectStruck, Length: 17631, dtype: object

In [24]:
crash_data[crash_data['IsAdditonalObjectStruck'] == False]['IsAdditonalObjectStruck']

49        False
122       False
137       False
249       False
258       False
          ...  
387969    False
388561    False
392354    False
392381    False
392813    False
Name: IsAdditonalObjectStruck, Length: 5372, dtype: object

Given that the majority of the values in IsAdditonalObjectStruck is NaN / Null and this was supposed to be only True or False and no additional information on what additional object means, it will be best to drop this column

In [25]:
crash_data = crash_data.drop(['IsAdditonalObjectStruck'], axis = 1)

In [26]:
crash_data.columns

Index(['Collision Id', 'Crash Date Time', 'City Name', 'County Code',
       'Collision Type Description', 'Day Of Week', 'HitRun',
       'IsHighwayRelated', 'IsTowAway', 'MotorVehicleInvolvedWithDesc',
       'MotorVehicleInvolvedWithOtherDesc', 'NumberInjured', 'NumberKilled',
       'Weather 1', 'Road Condition 1', 'LightingCode', 'LightingDescription',
       'Latitude', 'Longitude', 'PedestrianActionCode', 'PedestrianActionDesc',
       'Primary Collision Factor Code', 'Primary Collision Factor Violation',
       'PrimaryCollisionFactorIsCited', 'PrimaryCollisionPartyNumber',
       'PrimaryRoad', 'IsFreeway'],
      dtype='str')

In [27]:
crash_injuries = crash_data[['Collision Id','IsTowAway','MotorVehicleInvolvedWithDesc','MotorVehicleInvolvedWithOtherDesc','NumberInjured','NumberKilled',
'Primary Collision Factor Code', 'Primary Collision Factor Violation', 'PrimaryCollisionFactorIsCited', 'PrimaryCollisionPartyNumber']]

In [28]:
crash_injuries.head()

,Collision Id,IsTowAway,MotorVehicleInvolvedWithDesc,MotorVehicleInvolvedWithOtherDesc,NumberInjured,NumberKilled,Primary Collision Factor Code,Primary Collision Factor Violation,PrimaryCollisionFactorIsCited,PrimaryCollisionPartyNumber
0,4550265,True,NON-COLLISION,NaN,1.0,0.0,A,VC 22107,False,1.0
1,4550264,False,OTHER MOTOR VEHICLE,NaN,1.0,0.0,A,22350 VC,False,1.0
2,4550263,False,OTHER MOTOR VEHICLE,NaN,0.0,0.0,A,VC 22107,False,1.0
3,4550262,False,OTHER OBJECT,METAL OBJECT,0.0,0.0,A,VC 22350,False,1.0
4,4550261,True,OTHER MOTOR VEHICLE,NaN,3.0,0.0,A,VC 22450(a),False,1.0


In [29]:
crash_injuries

,Collision Id,IsTowAway,MotorVehicleInvolvedWithDesc,MotorVehicleInvolvedWithOtherDesc,NumberInjured,NumberKilled,Primary Collision Factor Code,Primary Collision Factor Violation,PrimaryCollisionFactorIsCited,PrimaryCollisionPartyNumber
0,4550265,True,NON-COLLISION,NaN,1.0,0.0,A,VC 22107,False,1.0
1,4550264,False,OTHER MOTOR VEHICLE,NaN,1.0,0.0,A,22350 VC,False,1.0
2,4550263,False,OTHER MOTOR VEHICLE,NaN,0.0,0.0,A,VC 22107,False,1.0
3,4550262,False,OTHER OBJECT,METAL OBJECT,0.0,0.0,A,VC 22350,False,1.0
4,4550261,True,OTHER MOTOR VEHICLE,NaN,3.0,0.0,A,VC 22450(a),False,1.0
...,...,...,...,...,...,...,...,...,...,...
393505,5105490,True,OTHER MOTOR VEHICLE,NaN,0.0,0.0,A,21801(a),False,1.0
393506,5105489,True,OTHER MOTOR VEHICLE,NaN,2.0,0.0,A,21453(a),False,1.0
393507,5105488,False,OTHER MOTOR VEHICLE,NaN,1.0,0.0,A,21801(a),False,1.0
393508,5105487,True,OTHER MOTOR VEHICLE,NaN,1.0,0.0,A,22350,False,1.0


In [30]:
crash_injuries['Primary Collision Factor Code'].value_counts()

Primary Collision Factor Code
A    361426
B     12369
C     10314
D      9137
Name: count, dtype: int64

It seems that for almost 92% of all reported accidents in CA in 2025, is the cause of some vehicle code being broken and not B - Other Improper Driving C - Other Than Driver D - Unknown E - Fell Asleep. Thus We should only be focusing on A 


In [31]:
primary_collision_factor_violation_values_A = crash_injuries[crash_injuries['Primary Collision Factor Code'] == 'A']

In [32]:
primary_collision_factor_violation_values_A

,Collision Id,IsTowAway,MotorVehicleInvolvedWithDesc,MotorVehicleInvolvedWithOtherDesc,NumberInjured,NumberKilled,Primary Collision Factor Code,Primary Collision Factor Violation,PrimaryCollisionFactorIsCited,PrimaryCollisionPartyNumber
0,4550265,True,NON-COLLISION,NaN,1.0,0.0,A,VC 22107,False,1.0
1,4550264,False,OTHER MOTOR VEHICLE,NaN,1.0,0.0,A,22350 VC,False,1.0
2,4550263,False,OTHER MOTOR VEHICLE,NaN,0.0,0.0,A,VC 22107,False,1.0
3,4550262,False,OTHER OBJECT,METAL OBJECT,0.0,0.0,A,VC 22350,False,1.0
4,4550261,True,OTHER MOTOR VEHICLE,NaN,3.0,0.0,A,VC 22450(a),False,1.0
...,...,...,...,...,...,...,...,...,...,...
393505,5105490,True,OTHER MOTOR VEHICLE,NaN,0.0,0.0,A,21801(a),False,1.0
393506,5105489,True,OTHER MOTOR VEHICLE,NaN,2.0,0.0,A,21453(a),False,1.0
393507,5105488,False,OTHER MOTOR VEHICLE,NaN,1.0,0.0,A,21801(a),False,1.0
393508,5105487,True,OTHER MOTOR VEHICLE,NaN,1.0,0.0,A,22350,False,1.0


In [33]:
primary_collision_factor_violation_values_A['Primary Collision Factor Violation'].value_counts()

Primary Collision Factor Violation
VC 22350                             55273
VC 22107                             37072
22350                                33788
22107                                31071
VC 21658(A)                          13432
                                     ...  
21806(c)                                 1
23152(a/b)                               1
22100(a2)                                1
VC 21651(B) DRV WRONG HWY INJ/DTH        1
29003(a)                                 1
Name: count, Length: 2554, dtype: int64

In [34]:
parsed_primary_collision_factor_violation_values_A = primary_collision_factor_violation_values_A['Primary Collision Factor Violation'].copy()

In [35]:
parsed_primary_collision_factor_violation_values_A = primary_collision_factor_violation_values_A['Primary Collision Factor Violation'].astype(str)

In [36]:
parsed_primary_collision_factor_violation_values_A

0            VC 22107
1            22350 VC
2            VC 22107
3            VC 22350
4         VC 22450(a)
             ...     
393505       21801(a)
393506       21453(a)
393507       21801(a)
393508          22350
393509          22350
Name: Primary Collision Factor Violation, Length: 361426, dtype: str

In [37]:
parsed_primary_collision_factor_violation_values_A = parsed_primary_collision_factor_violation_values_A.str.extract(r'(\d+)') # Series.str.extract(regex'(all decimals repeated)')

In [38]:
parsed_primary_collision_factor_violation_values_A

,0
0,22107
1,22350
2,22107
3,22350
4,22450
...,...
393505,21801
393506,21453
393507,21801
393508,22350


In [39]:
parsed_primary_collision_factor_violation_values_A = parsed_primary_collision_factor_violation_values_A.rename({0: 'Violation Code Number'}, axis = 'columns') 

In [40]:
parsed_primary_collision_factor_violation_values_A

,Violation Code Number
0,22107
1,22350
2,22107
3,22350
4,22450
...,...
393505,21801
393506,21453
393507,21801
393508,22350


parsed_primary_collision_factor_violation_values_A has all the traffic code violations filtered out. Now we need to find what is the distribution of this.

In [41]:
parsed_primary_collision_factor_violation_values_A.value_counts().head(5)

Violation Code Number
22350                    111508
22107                     82769
23152                     32700
21658                     29842
21453                     17256
Name: count, dtype: int64

Looking at the top 5 traffic violations involved in accidents, it seems to show that these 5 violations constitute over 75 percent of all accidents where some ticketable or worse offense occured.

These are what those traffic violations are: 

22350: Speeding

22107: Unsafe Lane Change

23152: DUI

21658: Not staying in Lane

21453: Red Light Violation

In [42]:
type(primary_collision_factor_violation_values_A)

pandas.DataFrame

In [43]:
primary_collision_factor_violation_values_A.insert(10, 'parsed_values', parsed_primary_collision_factor_violation_values_A) #Add the 10th column

In [44]:
primary_collision_factor_violation_values_A.drop('Primary Collision Factor Violation', axis = 1 )

,Collision Id,IsTowAway,MotorVehicleInvolvedWithDesc,MotorVehicleInvolvedWithOtherDesc,NumberInjured,NumberKilled,Primary Collision Factor Code,PrimaryCollisionFactorIsCited,PrimaryCollisionPartyNumber,parsed_values
0,4550265,True,NON-COLLISION,NaN,1.0,0.0,A,False,1.0,22107
1,4550264,False,OTHER MOTOR VEHICLE,NaN,1.0,0.0,A,False,1.0,22350
2,4550263,False,OTHER MOTOR VEHICLE,NaN,0.0,0.0,A,False,1.0,22107
3,4550262,False,OTHER OBJECT,METAL OBJECT,0.0,0.0,A,False,1.0,22350
4,4550261,True,OTHER MOTOR VEHICLE,NaN,3.0,0.0,A,False,1.0,22450
...,...,...,...,...,...,...,...,...,...,...
393505,5105490,True,OTHER MOTOR VEHICLE,NaN,0.0,0.0,A,False,1.0,21801
393506,5105489,True,OTHER MOTOR VEHICLE,NaN,2.0,0.0,A,False,1.0,21453
393507,5105488,False,OTHER MOTOR VEHICLE,NaN,1.0,0.0,A,False,1.0,21801
393508,5105487,True,OTHER MOTOR VEHICLE,NaN,1.0,0.0,A,False,1.0,22350


In [45]:
primary_collision_factor_violation_values_A['NumberKilled'].value_counts()

NumberKilled
 0.0    341278
 1.0      2144
 2.0       147
 3.0        18
 4.0         7
-1.0         1
 5.0         1
Name: count, dtype: int64

In [46]:
type(primary_collision_factor_violation_values_A['NumberKilled'][0])

numpy.float64

In [47]:
primary_collision_factor_violation_values_A.loc[primary_collision_factor_violation_values_A['NumberKilled'] == -1, 'NumberKilled'] = 0

In [48]:
primary_collision_factor_violation_values_A.loc[primary_collision_factor_violation_values_A['NumberKilled'].isna(), 'NumberKilled'] = 0

In [49]:
assert len(primary_collision_factor_violation_values_A['NumberKilled'].value_counts()) == primary_collision_factor_violation_values_A['NumberKilled'].nunique()  #confirm that all values are valid and there are no null values

In [50]:
primary_collision_factor_violation_values_A.loc[primary_collision_factor_violation_values_A['NumberInjured'] == -1, 'NumberInjured'] = 0

In [51]:
primary_collision_factor_violation_values_A.loc[primary_collision_factor_violation_values_A['NumberInjured'] < 0, 'NumberInjured'] = 0

In [52]:
primary_collision_factor_violation_values_A.loc[primary_collision_factor_violation_values_A['NumberInjured'].isna(), 'NumberInjured'] = 0

In [53]:
primary_collision_factor_violation_values_A['NumberInjured'].value_counts()

NumberInjured
0.0     207831
1.0     111044
2.0      29943
3.0       8511
4.0       2679
5.0        914
6.0        310
7.0        105
8.0         49
9.0         20
10.0        10
12.0         4
11.0         4
13.0         1
20.0         1
Name: count, dtype: int64

In [54]:
assert (primary_collision_factor_violation_values_A['NumberInjured'].nunique()) == len(primary_collision_factor_violation_values_A['NumberInjured'].value_counts()) 
#confirm that all values are valid and there are no null values

In [55]:
primary_collision_factor_violation_values_A['MotorVehicleInvolvedWithDesc'] = primary_collision_factor_violation_values_A['MotorVehicleInvolvedWithDesc'].fillna('N/A')

In [56]:
primary_collision_factor_violation_values_A['MotorVehicleInvolvedWithDesc'].value_counts()

MotorVehicleInvolvedWithDesc
OTHER MOTOR VEHICLE               235689
FIXED OBJECT                       59173
PARKED MOTOR VEHICLE               28377
PEDESTRIAN                         10565
BICYCLE                             9185
NON-COLLISION                       7081
OTHER OBJECT                        6620
MOTOR VEHICLE ON OTHER ROADWAY      3533
N/A                                  924
ANIMAL                               164
TRAIN                                115
Name: count, dtype: int64

In [57]:
primary_collision_factor_violation_values_A['MotorVehicleInvolvedWithDesc'] = primary_collision_factor_violation_values_A['MotorVehicleInvolvedWithDesc'].replace([0,'N/A'],'NOT-APPLICABLE')

In [58]:
primary_collision_factor_violation_values_A['MotorVehicleInvolvedWithDesc']

0               NON-COLLISION
1         OTHER MOTOR VEHICLE
2         OTHER MOTOR VEHICLE
3                OTHER OBJECT
4         OTHER MOTOR VEHICLE
                 ...         
393505    OTHER MOTOR VEHICLE
393506    OTHER MOTOR VEHICLE
393507    OTHER MOTOR VEHICLE
393508    OTHER MOTOR VEHICLE
393509           FIXED OBJECT
Name: MotorVehicleInvolvedWithDesc, Length: 361426, dtype: str

In [59]:
primary_collision_factor_violation_values_A['MotorVehicleInvolvedWithDesc'].value_counts()

MotorVehicleInvolvedWithDesc
OTHER MOTOR VEHICLE               235689
FIXED OBJECT                       59173
PARKED MOTOR VEHICLE               28377
PEDESTRIAN                         10565
BICYCLE                             9185
NON-COLLISION                       7081
OTHER OBJECT                        6620
MOTOR VEHICLE ON OTHER ROADWAY      3533
NOT-APPLICABLE                       924
ANIMAL                               164
TRAIN                                115
Name: count, dtype: int64

In [60]:
primary_collision_factor_violation_values_A['MotorVehicleInvolvedWithOtherDesc'] = primary_collision_factor_violation_values_A['MotorVehicleInvolvedWithOtherDesc'].fillna('N/A')

In [61]:
primary_collision_factor_violation_values_A['MotorVehicleInvolvedWithOtherDesc']

0                  N/A
1                  N/A
2                  N/A
3         METAL OBJECT
4                  N/A
              ...     
393505             N/A
393506             N/A
393507             N/A
393508             N/A
393509            CURB
Name: MotorVehicleInvolvedWithOtherDesc, Length: 361426, dtype: str

In [62]:
primary_collision_factor_violation_values_A

,Collision Id,IsTowAway,MotorVehicleInvolvedWithDesc,MotorVehicleInvolvedWithOtherDesc,NumberInjured,NumberKilled,Primary Collision Factor Code,Primary Collision Factor Violation,PrimaryCollisionFactorIsCited,PrimaryCollisionPartyNumber,parsed_values
0,4550265,True,NON-COLLISION,N/A,1.0,0.0,A,VC 22107,False,1.0,22107
1,4550264,False,OTHER MOTOR VEHICLE,N/A,1.0,0.0,A,22350 VC,False,1.0,22350
2,4550263,False,OTHER MOTOR VEHICLE,N/A,0.0,0.0,A,VC 22107,False,1.0,22107
3,4550262,False,OTHER OBJECT,METAL OBJECT,0.0,0.0,A,VC 22350,False,1.0,22350
4,4550261,True,OTHER MOTOR VEHICLE,N/A,3.0,0.0,A,VC 22450(a),False,1.0,22450
...,...,...,...,...,...,...,...,...,...,...,...
393505,5105490,True,OTHER MOTOR VEHICLE,N/A,0.0,0.0,A,21801(a),False,1.0,21801
393506,5105489,True,OTHER MOTOR VEHICLE,N/A,2.0,0.0,A,21453(a),False,1.0,21453
393507,5105488,False,OTHER MOTOR VEHICLE,N/A,1.0,0.0,A,21801(a),False,1.0,21801
393508,5105487,True,OTHER MOTOR VEHICLE,N/A,1.0,0.0,A,22350,False,1.0,22350


In [63]:
primary_collision_factor_violation_values_A['MotorVehicleInvolvedWithOtherDesc'].value_counts()

MotorVehicleInvolvedWithOtherDesc
N/A                          305009
TREE                           3361
GUARDRAIL                      1797
CONCRETE WALL                  1665
FENCE                          1543
                              ...  
FIRE HYDRANT #246                 1
CURB / FIRE HYDRANT / LIG         1
CURB(S) / TREE(S)                 1
LIGHT POST- CITY CURB             1
CITY SIGN/ CURB                   1
Name: count, Length: 9430, dtype: int64

In [64]:
primary_collision_factor_violation_values_A['MotorVehicleInvolvedWithOtherDesc'] 

0                  N/A
1                  N/A
2                  N/A
3         METAL OBJECT
4                  N/A
              ...     
393505             N/A
393506             N/A
393507             N/A
393508             N/A
393509            CURB
Name: MotorVehicleInvolvedWithOtherDesc, Length: 361426, dtype: str

In [65]:
primary_collision_factor_violation_values_A['MotorVehicleInvolvedWithOtherDesc'].isna()

0         False
1         False
2         False
3         False
4         False
          ...  
393505    False
393506    False
393507    False
393508    False
393509    False
Name: MotorVehicleInvolvedWithOtherDesc, Length: 361426, dtype: bool

In [66]:
primary_collision_factor_violation_values_A.loc[primary_collision_factor_violation_values_A['IsTowAway'].isna()]

,Collision Id,IsTowAway,MotorVehicleInvolvedWithDesc,MotorVehicleInvolvedWithOtherDesc,NumberInjured,NumberKilled,Primary Collision Factor Code,Primary Collision Factor Violation,PrimaryCollisionFactorIsCited,PrimaryCollisionPartyNumber,parsed_values
251921,4865348,NaN,PARKED MOTOR VEHICLE,N/A,0.0,0.0,A,22107,False,1.0,22107
253948,4862387,NaN,PEDESTRIAN,N/A,0.0,0.0,A,21950(a),False,1.0,21950
253949,4862385,NaN,BICYCLE,N/A,1.0,0.0,A,22107,False,1.0,22107
255034,4860884,NaN,BICYCLE,N/A,1.0,0.0,A,21801(a),False,1.0,21801
255038,4860872,NaN,OTHER MOTOR VEHICLE,N/A,0.0,0.0,A,22450(a),False,1.0,22450
...,...,...,...,...,...,...,...,...,...,...,...
380582,5096864,NaN,PARKED MOTOR VEHICLE,N/A,0.0,0.0,A,22107,False,1.0,22107
381596,5084731,NaN,OTHER MOTOR VEHICLE,N/A,1.0,0.0,A,21453(a),False,1.0,21453
388885,5040363,NaN,PEDESTRIAN,N/A,1.0,0.0,A,21950(a),False,NaN,21950
389254,5038819,NaN,BICYCLE,N/A,1.0,0.0,A,21800(A),False,1.0,21800


In [67]:
primary_collision_factor_violation_values_A.loc[primary_collision_factor_violation_values_A['IsTowAway'].isna(), 'IsTowAway'] = False

In [68]:
primary_collision_factor_violation_values_A.loc[primary_collision_factor_violation_values_A['IsTowAway'] == False]

,Collision Id,IsTowAway,MotorVehicleInvolvedWithDesc,MotorVehicleInvolvedWithOtherDesc,NumberInjured,NumberKilled,Primary Collision Factor Code,Primary Collision Factor Violation,PrimaryCollisionFactorIsCited,PrimaryCollisionPartyNumber,parsed_values
1,4550264,False,OTHER MOTOR VEHICLE,N/A,1.0,0.0,A,22350 VC,False,1.0,22350
2,4550263,False,OTHER MOTOR VEHICLE,N/A,0.0,0.0,A,VC 22107,False,1.0,22107
3,4550262,False,OTHER OBJECT,METAL OBJECT,0.0,0.0,A,VC 22350,False,1.0,22350
5,4550260,False,OTHER MOTOR VEHICLE,N/A,0.0,0.0,A,VC 22350,False,1.0,22350
10,4550253,False,PARKED MOTOR VEHICLE,N/A,0.0,0.0,A,22350,NaN,1.0,22350
...,...,...,...,...,...,...,...,...,...,...,...
393497,5105499,False,OTHER MOTOR VEHICLE,N/A,1.0,0.0,A,22350,False,1.0,22350
393498,5105498,False,OTHER MOTOR VEHICLE,N/A,1.0,0.0,A,22350,False,1.0,22350
393499,5105497,False,PEDESTRIAN,N/A,1.0,0.0,A,22350,False,1.0,22350
393504,5105492,False,BICYCLE,N/A,0.0,0.0,A,21802(a),False,1.0,21802


In [69]:
primary_collision_factor_violation_values_A.to_csv("Main Collision Factors.csv")

In [70]:
crash_parties_data

,_id,PartyId,CollisionId,PartyNumber,PartyType,IsAtFault,IsOnDutyEmergencyVehicle,IsHitAndRun,AirbagCode,AirbagDescription,...,Vehicle2TypeDesc,Vehicle2Year,Vehicle2Make,Vehicle2Model,Vehicle2Color,V2IsVehicleTowed,Lane,ThruLanes,TotalLanes,IsDREConducted
0,1,13900346,4541902,1,Driver,True,False,True,B,UNKNOWN,...,NaN,NaN,NaN,NaN,NaN,NaN,N/B,1.0,1.0,False
1,2,13900345,4541901,2,Driver,NaN,False,False,M,AIR BAG NOT DEPLOYED,...,NaN,NaN,NaN,NaN,NaN,NaN,2,2.0,5.0,False
2,3,13900344,4541901,1,Driver,True,False,False,L,AIR BAG DEPLOYED,...,NaN,NaN,NaN,NaN,NaN,NaN,2,2.0,5.0,False
3,4,13900343,4541900,2,Driver,NaN,False,False,M,AIR BAG NOT DEPLOYED,...,NaN,NaN,NaN,NaN,NaN,NaN,E/B,1.0,1.0,False
4,5,13900342,4541900,1,Driver,NaN,False,False,M,AIR BAG NOT DEPLOYED,...,NaN,NaN,NaN,NaN,NaN,NaN,N/B,1.0,1.0,False
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
767047,767048,14979609,5105674,1,Driver,True,False,NaN,M,AIR BAG NOT DEPLOYED,...,NaN,NaN,NaN,NaN,NaN,NaN,1,1.0,1.0,NaN
767048,767049,14979608,5105673,1,Driver,True,False,NaN,M,AIR BAG NOT DEPLOYED,...,NaN,NaN,NaN,NaN,NaN,NaN,1,1.0,1.0,NaN
767049,767050,14979607,5105672,2,Driver,NaN,False,NaN,M,AIR BAG NOT DEPLOYED,...,NaN,NaN,NaN,NaN,NaN,NaN,1,3.0,3.0,NaN
767050,767051,14979606,5105672,1,Driver,True,False,NaN,M,AIR BAG NOT DEPLOYED,...,NaN,NaN,NaN,NaN,NaN,NaN,2,3.0,3.0,NaN


In [71]:
crash_parties_data.drop(crash_parties_data[crash_parties_data['Vehicle1Year'] < 1980].index,inplace = True) 

In [72]:
#crash_parties_data.dropna(subset = ['Vehicle2Year'],inplace = True) 

In [73]:
crash_parties_data['IsAtFault']

0         True
1          NaN
2         True
3          NaN
4          NaN
          ... 
767047    True
767048    True
767049     NaN
767050    True
767051     NaN
Name: IsAtFault, Length: 765797, dtype: object

In [74]:
crash_parties_data['PartyType'].value_counts()

PartyType
Driver           682445
ParkedVehicle     54432
Pedestrian        13114
Bicyclist         11723
Other              3816
Operator            267
Name: count, dtype: int64

In [75]:
crash_parties_data['SafetyEquipmentCode'].value_counts()

SafetyEquipmentCode
G    549632
B     46037
P     13245
W     10668
C      9588
H      6789
A      5383
E      3151
V      2513
F      1331
D       741
J       513
Y       199
Q       124
X       120
K        62
S        62
U        56
R        29
T        16
L        12
N         1
Name: count, dtype: int64

It seems that in about 80 percent of reported accidents, people do use their seatbelts.

In [76]:
crash_parties_data['SafetyEquipmentDescription'].value_counts()

SafetyEquipmentDescription
OCCUPANT LAP SHOULDER HARNESS USED         549632
OCCUPANT UNKNOWN                            46037
NOT REQUIRED                                13245
MOTOR BICYCLE DRIVER WITH HELMET            10668
OCCUPANT LAP BELT USED                       9588
OCCUPANT LAP SHOULDER HARNESS NOT USED       6789
OCCUPANT NONE IN VEHICLE                     5383
OCCUPANT SHOULDER HARNESS USED               3151
MOTOR BICYCLE DRIVER NO HELMET               2513
OCCUPANT SHOULDER HARNESS NOT USED           1331
OCCUPANT LAP BELT NOT USED                    741
OCCUPANT PASSIVE RESTRAINT USED               513
MOTOR BICYCLE PASSENGER WITH HELMET           199
CHILD IN VEHICLE USED                         124
MOTOR BICYCLE PASSENGER NO HELMET             120
OCCUPANT PASSIVE RESTRAINT NOT USED            62
CHILD IN VEHICLE USE UNKNOWN                   62
CHILD RESTRAINT NONE IN VEHICLE                56
CHILD IN VEHICLE NOT USED                      29
CHILD RESTRAINT IN VEHI

In [77]:
crash_parties_data['Special Information'].value_counts()

Special Information
CELL PHONE NOT IN USE                                  603107
CELL PHONE USE UNKNOWN                                  89882
CELL PHONE HANDHELD IN USE                               4944
CELL PHONE HANDSFREE IN USE                              3372
HAZARDOUS MATERIAL                                        589
CELL PHONE NOT IN USE / SCHOOL BUS RELATED                440
HAZARDOUS MATERIAL / CELL PHONE NOT IN USE                313
SCHOOL BUS RELATED                                        291
CELL PHONE USE UNKNOWN / SCHOOL BUS RELATED                46
CELL PHONE NOT IN USE / CELL PHONE USE UNKNOWN             46
HAZARDOUS MATERIAL / CELL PHONE USE UNKNOWN                20
CELL PHONE HANDSFREE IN USE / CELL PHONE NOT IN USE         7
HAZARDOUS MATERIAL / CELL PHONE HANDHELD IN USE             5
CELL PHONE HANDHELD IN USE / CELL PHONE NOT IN USE          3
HAZARDOUS MATERIAL / SCHOOL BUS RELATED                     3
HAZARDOUS MATERIAL / CELL PHONE HANDSFREE IN USE  

It seems that cell phone usage when driving might not be a reliable indicator for determing accidents.

In [78]:
crash_parties_data['Other Associate Factor'].value_counts()

Other Associate Factor
NONE APPARENT                                                      647223
VC SECTION VIOLATED - VC 22107                                      10497
VC SECTION VIOLATED                                                 10163
INATTENTION                                                          8846
STOP & GO TRAFFIC                                                    8252
                                                                    ...  
VC SECTION VIOLATED - 21651(a)(1) / OTHER                               1
VC SECTION VIOLATED - 14601.1(a) / NONE APPARENT                        1
VC SECTION VIOLATED - 23123(a) / INATTENTION                            1
VC SECTION VIOLATED - 23152(f) / NONE APPARENT                          1
VC SECTION VIOLATED - 26709(a)(1) / DEFECTIVE VEHICLE EQUIPMENT         1
Name: count, Length: 3653, dtype: int64

In [79]:
crash_parties_data['Inattention'].value_counts()

Inattention
OTHER                   6129
CELL PHONE HANDHELD     1200
CELL PHONE HANDSFREE     530
ELECTRONIC_EQUIPMENT     408
EATING                   343
RADIO/CD                 340
CHILDREN                 299
ANIMALS                  194
PERSONAL HYGIENE          56
SMOKING                   41
READING                   24
Name: count, dtype: int64

It seems that people who are distracted when driving might be distracted by something in their enviornment (not cellphones, music, passengers)

In [80]:
crash_parties_data['SpeedLimit'].value_counts().head(10)

SpeedLimit
65.0    211033
55.0     78802
25.0     71706
35.0     63725
45.0     43642
40.0     43167
30.0     16560
50.0     16534
70.0     14009
15.0      5830
Name: count, dtype: int64

It seems that the majority of accidents occur on streets ranging between 15-65 mph with nearly half being around roads on 55-65 mph road. 

In [81]:
crash_parties_car_data = crash_parties_data[['CollisionId', 'Vehicle1Year',
       'Vehicle1Make', 'Vehicle1Model', 'Vehicle1Color', 'V1IsVehicleTowed']]

In [82]:
Vehicle2 = crash_parties_data[['CollisionId', 'Vehicle2Year', 'Vehicle2Make', 'Vehicle2Model', 'Vehicle2Color', 'V2IsVehicleTowed']]

In [83]:
Vehicle2 = Vehicle2.rename(columns = {'Vehicle2Year': 'Vehicle1Year', 'Vehicle2Make': 'Vehicle1Make', 
                            'Vehicle2Model':'Vehicle1Model','Vehicle2Color':'Vehicle1Color','V2IsVehicleTowed' : 'V1IsVehicleTowed'})

In [84]:
Vehicle2

,CollisionId,Vehicle1Year,Vehicle1Make,Vehicle1Model,Vehicle1Color,V1IsVehicleTowed
0,4541902,NaN,NaN,NaN,NaN,NaN
1,4541901,NaN,NaN,NaN,NaN,NaN
2,4541901,NaN,NaN,NaN,NaN,NaN
3,4541900,NaN,NaN,NaN,NaN,NaN
4,4541900,NaN,NaN,NaN,NaN,NaN
...,...,...,...,...,...,...
767047,5105674,NaN,NaN,NaN,NaN,NaN
767048,5105673,NaN,NaN,NaN,NaN,NaN
767049,5105672,NaN,NaN,NaN,NaN,NaN
767050,5105672,NaN,NaN,NaN,NaN,NaN


In [85]:
Vehicle2['V1IsVehicleTowed']= Vehicle2['V1IsVehicleTowed'].fillna(False)

In [86]:
Vehicle2 = Vehicle2.dropna()

In [87]:
crash_parties_car_data = pd.concat([crash_parties_car_data, Vehicle2])

In [88]:
crash_parties_car_data

,CollisionId,Vehicle1Year,Vehicle1Make,Vehicle1Model,Vehicle1Color,V1IsVehicleTowed
0,4541902,NaN,NaN,NaN,NaN,False
1,4541901,2005.0,FORD,F150,GRY,True
2,4541901,2002.0,FORD,TAURUS,GLD,True
3,4541900,2001.0,STRN,ION,GLD,False
4,4541900,2020.0,KIA,OPTIMA,WHI,False
...,...,...,...,...,...,...
765984,5107065,2024.0,GREAT DANE TRAILERS,TRUCK-TRACTOR,WHITE,False
766193,5106155,2024.0,GDAN,TRAILER,WHITE,False
766407,5106033,2025.0,ALKOTA,2 AXLE TRLR,BLACK,False
766490,5105989,1995.0,TREIN,CARRIER,BLACK,False


In [89]:
crash_parties_car_data['V1IsVehicleTowed']= crash_parties_car_data['V1IsVehicleTowed'].fillna(False)

In [90]:
crash_parties_car_data['Vehicle1Make'] = crash_parties_car_data['Vehicle1Make'].str.upper().str.strip()

In [91]:
crash_parties_car_data['Vehicle1Make'].value_counts().head(20)

Vehicle1Make
TOYT         71943
FORD         67648
TOYOTA       58568
HOND         47408
HONDA        41707
CHEV         30798
CHEVROLET    24599
NISS         23151
NISSAN       20944
KIA          20652
BMW          18157
GMC          14064
JEEP         13757
HYUN         13038
DODG         10694
DODGE        10417
LEXS         10249
HYUNDAI      10015
MERZ          9335
FRHT          8822
Name: count, dtype: int64

In [92]:
crash_parties_car_data = crash_parties_car_data.replace(['TOYT','Toyota','PRUIS','SEQUIA','SEQUIOA', 'TOYOTA /', 
                                                         'TOYTA', 'TOYOA','TOYO','TOY','TOTY','TOTYA','TOYOT','TOYOTA TACOMA','TOYOTA015295',
                                                        'TOYOTAA','TOYOTOA','TOYOTQ','TOYOTS','TOYOYTA','TOYP','TOYT-TOYOTA','TOYTA','TOYTOA','TOYTT','TYOTA'
                                                        ],'TOYOTA')

In [93]:
crash_parties_car_data = crash_parties_car_data.replace(['HOND','Honda','HONDA ODISSEY','HONDA /', 'HONA','HOINDA','HOND-HONDA','HONE'],'HONDA')

In [94]:
crash_parties_car_data = crash_parties_car_data.replace(['CHEV','CHEVY','CHREVROLET','Chevrolet', 'CHEVROLET /','CEHVROLET','CHEV-CHEVROLET',
                                                         'CHEVERLOT','CHEVEROLET', 'CHEVORLET','CHEVRELOT','CHEVROLT','CHRVROLET','CHEVEROLET'],'CHEVROLET')

In [95]:
crash_parties_car_data = crash_parties_car_data.replace(['NISS','Nissan'],'NISSAN')

In [96]:
crash_parties_car_data = crash_parties_car_data.replace(['F0RD','F-150','FORD-FORD','FORSD'],'FORD')

In [97]:
crash_parties_car_data = crash_parties_car_data.replace(['HYUN','HYUN - HYUNDAI','HYUNDAU'],'HYUNDAI')

In [98]:
crash_parties_car_data = crash_parties_car_data.replace(['KIA MOTOR CORPORATION'],'KIA')

In [99]:
crash_parties_car_data = crash_parties_car_data.replace(['DODG','DODGEDODG', 'DODGE /'],'DODGE')

In [100]:
crash_parties_car_data = crash_parties_car_data.replace('LEXS','LEXUS')

In [101]:
crash_parties_car_data = crash_parties_car_data.replace(['MERZ','MERCEDES','MERC','MERCEDEZ','MBZ', 'MERCEDES BENZ'],'MERCEDES-BENZ')

In [102]:
crash_parties_car_data = crash_parties_car_data.replace('SUBA','SUBARU')

In [103]:
crash_parties_car_data = crash_parties_car_data.replace(['MAZD','MADA'],'MAZDA')

In [104]:
crash_parties_car_data = crash_parties_car_data.replace(['TESL','TSMR', 'TESLA MOTORS'],'TESLA')

In [105]:
crash_parties_car_data = crash_parties_car_data.replace(['VOLK','VOLKSWAGON','VW'],'VOLKSWAGEN')

In [106]:
crash_parties_car_data = crash_parties_car_data.replace('ACUR','ACURA')

In [107]:
crash_parties_car_data = crash_parties_car_data.replace(['INFI','INFINITY'],'INFINITI')

In [108]:
crash_parties_car_data = crash_parties_car_data.replace('VOLV','VOLVO')

In [109]:
crash_parties_car_data = crash_parties_car_data.replace(['FRHT','PTRB','KW','FREIGHTLINER','MACK  TR  BLUE','MACK',
                                                         'PETERBILT','HINO','KENWORTH','PETERBUILT', 'PTBR', 'PTR', 'PB', 'PETERBILT MOTOR CO.', 'PERTERBILT', 'PETER'
                                                        'PENSKE'],'SEMI-TRUCK')

In [110]:
crash_parties_car_data = crash_parties_car_data.replace(['CHRY','CHYRSLER','CHRYSTLER','CHRY - CHRYSLER','CRYSLER', 'CHRYS','CHYSLER'], 'CHRYSLER')

In [111]:
crash_parties_car_data = crash_parties_car_data.replace(['JEEP (1989 AND LATER'], 'JEEP')

In [112]:
crash_parties_car_data = crash_parties_car_data.replace('BUIC','BUICK')

In [113]:
crash_parties_car_data = crash_parties_car_data.replace('CADI','CADILLAC')

In [114]:
crash_parties_car_data = crash_parties_car_data.replace('HD','HARLEY DAVIDSON')

In [115]:
crash_parties_car_data = crash_parties_car_data.replace(['MITS','MITSUBISH'],'MITSUBISHI')

In [116]:
crash_parties_car_data = crash_parties_car_data.replace('YAMA','YAMAHA')

In [117]:
crash_parties_car_data = crash_parties_car_data.replace(['SCIO','SION'],'SCION')

In [118]:
crash_parties_car_data = crash_parties_car_data.replace('KAWK','KAWAKSAKI')

In [119]:
crash_parties_car_data = crash_parties_car_data.replace('LNDR','LAND ROVER')

In [120]:
crash_parties_car_data = crash_parties_car_data.replace('LINC','LINCOLN')

In [121]:
crash_parties_car_data = crash_parties_car_data.replace(['MNNI','MINI'],'MINI COOPER')

In [122]:
crash_parties_car_data = crash_parties_car_data.replace('SUZI','SUZUKI')

In [123]:
crash_parties_car_data = crash_parties_car_data.replace(['FORD TAURUS','Ford','FORD ', 'FORD /'],'FORD')

In [124]:
crash_parties_car_data = crash_parties_car_data.replace('PONT','PONTIAC')

In [125]:
crash_parties_car_data = crash_parties_car_data.replace('STRN','SATURN')

In [126]:
crash_parties_car_data = crash_parties_car_data.replace('JAGU','JAGUAR')

In [127]:
crash_parties_car_data = crash_parties_car_data.replace('ISU','ISUZU')

In [128]:
crash_parties_car_data = crash_parties_car_data.replace('PORS','PORSCHE')

In [129]:
crash_parties_car_data = crash_parties_car_data.replace(['UTIL','UTILITY TRAILER','UTILI', 'STRIPER' ],'UTILITY')

In [130]:
crash_parties_car_data = crash_parties_car_data.replace(['HUMMER','GENERAL MOTORS CORP.','HUMM'],'GMC')

In [131]:
crash_parties_car_data = crash_parties_car_data.replace(['RIVI'],'RIVIAN')

In [132]:
crash_parties_car_data = crash_parties_car_data.replace(['BLUEBIRD','BB','NEW FLYER','BLUE BIRD', 'OTHER - BUS', 'GILLIG (BUS)', 'GILLIG', 'NABI', 'THOB','GREAT DANE TRAILERS'], 'BUS')

In [133]:
crash_parties_car_data = crash_parties_car_data.replace(['MASE'],'MASERATI')

In [134]:
crash_parties_car_data = crash_parties_car_data.replace(['SCHWINN','BIKE'],'BICYCLE')

In [135]:
crash_parties_car_data = crash_parties_car_data.replace(['WABASH'],'TRAIN')

In [136]:
crash_parties_car_data = crash_parties_car_data.replace(['WANC', 'GREAT DANE','GDAN', 'UNK','HYTR', 'U','HYUNDAI TRANSLEAD TRAILER'],'TRAILER')

In [137]:
crash_parties_car_data = crash_parties_car_data.replace(['INTL'],'INTERNATIONAL')

In [138]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'M3') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'HONDA'), 'Vehicle1Make'] = 'BMW'

In [139]:
crash_parties_car_data = crash_parties_car_data.replace('/', '')

In [140]:
crash_parties_car_data['Vehicle1Make'].value_counts().head(50).sum()

np.int64(689459)

In [141]:
crash_parties_car_data['Vehicle1Make'].value_counts().head(60)

Vehicle1Make
TOYOTA             130785
HONDA               89239
FORD                67725
CHEVROLET           56553
NISSAN              44095
HYUNDAI             23056
DODGE               21135
KIA                 20751
TESLA               19074
LEXUS               18771
SEMI-TRUCK          18743
BMW                 18158
MERCEDES-BENZ       17829
GMC                 14357
JEEP                13758
SUBARU              11815
MAZDA               11585
VOLKSWAGEN          11082
ACURA                8637
RAM                  6514
AUDI                 6413
INFINITI             6291
CHRYSLER             5067
VOLVO                4417
CADILLAC             4277
BUICK                2689
HARLEY DAVIDSON      2652
MITSUBISHI           2634
TRAILER              2602
INTERNATIONAL        2594
UTILITY              2179
SCION                2083
YAMAHA               2062
UNKNOWN              1907
LAND ROVER           1857
BUS                  1845
LINCOLN              1837
PORSCHE              1680

In [142]:
crash_parties_car_data['Vehicle1Make'].value_counts()[60:120]

Vehicle1Make
SPECIALIZED                185
SPCNS                      184
AUTOCAR                    182
INTERNATIONAL HARVESTER    154
BEALL                      153
ELDOR                      148
VC SECTION 20002(A)        135
POLARIS                    135
TRIU                       133
DUCATI                     133
ALFA ROMEO                 132
GNSI                       130
VANGUARD                   128
HUFFY                      127
WABA                       123
THMS                       120
FREIGHT LINER              120
WABAS                      119
POLESTAR                   112
ALFA                       111
JOHN DEERE                 111
HEIL                       110
PIERCE                     109
OTHER - TRUCK              105
AUTC                       102
SEGWAY                     102
LUCID                      102
WESTERN STAR                98
TRIUMPH                     98
BIGTX                       97
GILG                        96
ACE                       

In [143]:
crash_parties_car_data.loc[crash_parties_car_data['Vehicle1Year'] > 2026, 'Vehicle1Year'] = 2025

In [144]:
crash_parties_car_data.loc[crash_parties_car_data['Vehicle1Year'] > 2026, 'Vehicle1Year'] = 2025

In [145]:
crash_parties_car_data['Vehicle1Model'].value_counts()[51:100]

Vehicle1Model
YUKON             2071
FORESTER          2049
GRAND CHEROKEE    1924
SOUL              1876
2500              1850
RX350             1831
CAMARO            1830
SUBURBAN          1827
COLORADO          1822
SORENTO           1766
CX-5              1731
CRUZE             1709
MDX               1689
RAM 1500 PU       1680
TRAVERSE          1658
CHALLENGER        1653
DURANGO           1650
EDGE              1627
TRANSIT           1617
SANTA FE          1569
F350              1541
AVALON            1527
PATHFINDER        1519
FIT               1511
CROSSTREK         1509
EXPEDITION        1506
TL                1425
328I              1409
SPORTAGE          1367
ES350             1336
C300              1321
300               1278
IMPALA            1242
YARIS             1226
IS250             1201
EXPRESS           1158
X5                1155
MAZDA3            1143
RAM               1127
X3                1111
COMPASS           1096
G35               1076
BUS               10

In [146]:
crash_parties_car_data['Vehicle1Model'] = crash_parties_car_data['Vehicle1Model'].str.upper().str.strip()

In [147]:
crash_parties_car_data = crash_parties_car_data.replace(['3','Model 3', 'MODEL 3 '],'MODEL 3')

In [148]:
crash_parties_car_data = crash_parties_car_data.replace(['Y', 'Model Y', 'MODEL Y ', 'Y SERIES'], 'MODEL Y')

In [149]:
crash_parties_car_data = crash_parties_car_data.replace(['CR-V'], 'CRV')

In [150]:
crash_parties_car_data = crash_parties_car_data.replace(['F150'], 'F-150')

In [151]:
crash_parties_car_data = crash_parties_car_data.replace(['F250'], 'F-250')

In [152]:
crash_parties_car_data = crash_parties_car_data.replace(['RAM 1500 PU', '1500'], 'RAM 1500')

In [153]:
crash_parties_car_data = crash_parties_car_data.replace(['RAM 2500 PU', '2500', '2500 HD'], 'RAM 2500')

In [154]:
crash_parties_car_data = crash_parties_car_data.replace(['RAM 3500 PU', '3500'], 'RAM 3500')

In [155]:
crash_parties_car_data = crash_parties_car_data.replace(['RAV-4', 'RAV4 HYBRID', 'Rav 4', 'RAV 4', 'RAV'], 'RAV4')

In [156]:
crash_parties_car_data = crash_parties_car_data.replace(['4RUNNER', '4 RUNNER', '4 Runner', '4-Runner ', '4-Runner', '4RN' ], '4-RUNNER')

In [157]:
crash_parties_car_data = crash_parties_car_data.replace(['S','s', 'Model S'], 'MODEL S')

In [158]:
crash_parties_car_data = crash_parties_car_data.replace(['  CASCADIA', ' CASCADIA'], 'CASCADIA')

In [159]:
crash_parties_car_data = crash_parties_car_data.replace(['MAZDA 3'], 'MAZDA3')

In [160]:
crash_parties_car_data = crash_parties_car_data.replace(['X'], 'MODEL X')

In [161]:
crash_parties_car_data = crash_parties_car_data.replace(['Civic','CIVIC (AND CRX)', ' CIVIC'], 'CIVIC')

In [162]:
crash_parties_car_data = crash_parties_car_data.replace(['Accord', ' Accord', ' ACCORD'], 'ACCORD')

In [163]:
crash_parties_car_data = crash_parties_car_data.replace(['Camry', ' Camry', ' CAMRY','CAM','CAMARY','CAMRY 4D', 'CAMRY /', 'CAMRY SE'], 'CAMRY')

In [164]:
crash_parties_car_data = crash_parties_car_data.replace(['Corolla','COA'], 'COROLLA')

In [165]:
crash_parties_car_data = crash_parties_car_data.replace(['HGH', 'HHR'], 'HIGHLANDER')

In [166]:
crash_parties_car_data = crash_parties_car_data.replace(['SNA','VAN'], 'SIENNA')

In [167]:
crash_parties_car_data = crash_parties_car_data.replace(['PRI', 'Prius'], 'PRIUS')

In [168]:
crash_parties_car_data = crash_parties_car_data.replace(['ES300H'], 'ES300')

In [169]:
crash_parties_car_data = crash_parties_car_data.replace(['Silverado'], 'SILVERADO')

In [170]:
crash_parties_car_data = crash_parties_car_data.replace(['Tacoma', 'TAC'], 'TACOMA')

In [171]:
crash_parties_car_data = crash_parties_car_data.replace(['TUN'], 'TUNDRA')

In [172]:
crash_parties_car_data = crash_parties_car_data.replace(['M-3', '3 Series','330I','328I','325I','330CI','320I','335I', '330', '340I'], '3 SERIES')

In [173]:
crash_parties_car_data = crash_parties_car_data.replace(['F550'], 'F-550')

In [174]:
crash_parties_car_data = crash_parties_car_data.replace(['F350'], 'F-350')

In [175]:
crash_parties_car_data = crash_parties_car_data.replace(['Ariya'], 'ARIYA')

In [176]:
crash_parties_car_data = crash_parties_car_data.replace(['IONIQ5'], 'IONIQ 5')

In [177]:
crash_parties_car_data = crash_parties_car_data.replace(['TRUCK TRACTOR- GASOLINE', 'TRUCK TRACTOR- DIESEL'], 'TRUCK TRACTOR')

In [178]:
crash_parties_car_data = crash_parties_car_data.replace(['BIKE'], 'BICYCLE')

In [179]:
crash_parties_car_data = crash_parties_car_data.replace(['IONIC'], 'IONIQ')

In [180]:
crash_parties_car_data = crash_parties_car_data.replace(['TOWN AND COUNTRY'], 'TOWN & COUNTRY')

In [181]:
crash_parties_car_data = crash_parties_car_data.replace(['ID .4', 'ID.4'], 'ID4')

In [182]:
crash_parties_car_data = crash_parties_car_data.replace(['F450'], 'F-450')

In [183]:
crash_parties_car_data = crash_parties_car_data.replace(['MAZDA 6', 'MAZDA 6 '], 'MAZDA6')

In [184]:
crash_parties_car_data = crash_parties_car_data.replace(['CAYENNE S'], 'CAYENNE')

In [185]:
crash_parties_car_data = crash_parties_car_data.replace(['E-350', 'ECONOLINE E350'], 'ECONOLINE E350')

In [186]:
crash_parties_car_data = crash_parties_car_data.replace(['CYBER TRUCK' ], 'CYBERTRUCK')

In [187]:
crash_parties_car_data = crash_parties_car_data.replace(['IS 250', 'IS200T' ], 'IS250')

In [188]:
crash_parties_car_data = crash_parties_car_data.replace(['SEMITRUCK','SEMI', 'FTL', 'FREIGHTLNR', 'FRT', 'INTERNATIONAL', 'KRMCT', 'PENSKE' ], 'SEMI-TRUCK')

In [189]:
crash_parties_car_data = crash_parties_car_data.replace(['HRV'], 'HR-V')

In [190]:
crash_parties_car_data = crash_parties_car_data.replace(['CHR'], 'CH-R')

In [191]:
crash_parties_car_data = crash_parties_car_data.replace(['C-CLASS'], 'C350')

In [192]:
crash_parties_car_data = crash_parties_car_data.replace(['CMAX'], 'C-MAX')

In [193]:
crash_parties_car_data = crash_parties_car_data.replace(['RX450H'], 'RX450')

In [194]:
crash_parties_car_data = crash_parties_car_data.replace(['NX350H'], 'NX350')

In [195]:
crash_parties_car_data = crash_parties_car_data.replace(['UTILITY TRAILER', 'BOX TRAILER', 'SEMI TRAILER', 'DUMP TRAILER', 'FLATBED TRAILER', 'TANK TRAILER', 'TRLR'], 'TRAILER')

In [196]:
crash_parties_car_data = crash_parties_car_data.replace(['DUMP'],'DUMP TRUCK')

In [197]:
crash_parties_car_data = crash_parties_car_data.replace(['300, CHRYSLER 300', '300 Series'],'300 SERIES')

In [198]:
crash_parties_car_data = crash_parties_car_data.replace(['GLE 350'],'GLE350')

In [199]:
crash_parties_car_data = crash_parties_car_data.replace(['X-TERRA'],'XTERRA')

In [200]:
crash_parties_car_data = crash_parties_car_data.replace(['CX30'],'CX-30')

In [201]:
crash_parties_car_data = crash_parties_car_data.replace(['CX50'],'CX-50')

In [202]:
crash_parties_car_data = crash_parties_car_data.replace(['GLC300'],'GLC 300')

In [203]:
crash_parties_car_data = crash_parties_car_data.replace(['KICK'],'KICKS')

In [204]:
crash_parties_car_data = crash_parties_car_data.replace(['R1 S', 'RS1', 'R15','RIS','R1S ADVENTURE', 'RS','R1-S','S1','R2S','RS-1','R/S'],'R1S')

In [205]:
crash_parties_car_data = crash_parties_car_data.replace(['R17', 'R1F', 'R1 T', 'RT1','R IT', 'RIVIAN R1T','T1','R1T LAUNCH EDITION', 'R-1T', 'RIT'],'R1T')

In [206]:
crash_parties_car_data = crash_parties_car_data.replace(['SCHOOL BUS'],'BUS') #Putting together the Bus categories

In [207]:
crash_parties_car_data = crash_parties_car_data.replace(['NINJA'],'MOTORCYCLE') #Putting together the Bus categories

In [208]:
crash_parties_car_data = crash_parties_car_data.replace(['EXPRESS', 'SPRINTER', 'TRANSIT 350', 'EDV 700', 'EDV TRIM 700', 'EDV','TRANSPORT', 
                                                         'RCV','AMAZON','EDV700', 'CARAVAN', 'EDU700','700', 'UTILY','U-HAUL'],'UTILITY VAN') 
#EXPRESS Vehicles are essentially those white utility vans

In [209]:
crash_parties_car_data = crash_parties_car_data.replace('MRAI','Mirai') 
#EXPRESS Vehicles are essentially those white utility vans

In [210]:
crash_parties_car_data = crash_parties_car_data.replace(['S-10'],'S10') #EXPRESS Vehicles are essentially those white utility vans

In [211]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == '6') 
                            & (crash_parties_car_data['Vehicle1Make'] == 'TESLA'), 'Vehicle1Model'] = 'MODEL 3' 

In [212]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == '6') 
                            & (crash_parties_car_data['Vehicle1Make'] == 'MAZDA'), 'Vehicle1Model'] = 'MAZDA6'

In [213]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == '300') 
                            & (crash_parties_car_data['Vehicle1Make'] == 'CHRYSLER'), 'Vehicle1Model'] = '300 Series' 

In [214]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == '300') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'LEXUS') & 
                            (crash_parties_car_data['Vehicle1Year'] > 2016), 'Vehicle1Model'] = 'IS300'

In [215]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == '300') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'LEXUS')&
                            (crash_parties_car_data['Vehicle1Year'] < 2001), 'Vehicle1Model'] = 'RX300'

In [216]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == '300') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'LEXUS')&
                            (crash_parties_car_data['Vehicle1Year'] > 2003), 'Vehicle1Model'] = 'IS300'

In [217]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == '300') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'LEXUS'),'Vehicle1Model'] = 'IS300'
                            

In [218]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == '200') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'LEXUS'),'Vehicle1Model'] = 'IS200'
                            

In [219]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == '200') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'BMW'),'Vehicle1Model'] = '2 SERIES'
                            

In [220]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == '200') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'CHEVROLET'),'Vehicle1Make'] = 'CHRYSLER'
                            

In [221]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == '300') &
                            (crash_parties_car_data['Vehicle1Make'] == 'MERCEDES-BENZ'), 'Vehicle1Model'] = 'C300'

In [222]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'M3') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'TESLA'), 'Vehicle1Model'] = 'MODEL 3'

In [223]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'M3') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'BMW'), 'Vehicle1Model'] = '3 SERIES'

In [224]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'M3') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'MAZDA'), 'Vehicle1Model'] = 'MAZDA3'

In [225]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'].isin(['PICK UP', 'PICKUP']))  & 
                            (crash_parties_car_data['Vehicle1Make'] == 'TOYOTA'), 'Vehicle1Model'] = 'TACOMA' #Labeling it as Tacoma since it seems that pick up trucks are smaller

In [226]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'TRUCK') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'TOYOTA'), 'Vehicle1Model'] = 'TACOMA' #Labeling it as Tacoma since it seems that pick up trucks are smaller

In [227]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'PICK UP') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'GMC'),'Vehicle1Model'] = 'Sierra' #Labeling it as Sierra since it was most popular GMC truck in this class

In [228]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'TRUCK') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'GMC'),'Vehicle1Model'] = 'Sierra' #Labeling it as Sierra since it was most popular GMC truck in this class

In [229]:
crash_parties_car_data.drop(crash_parties_car_data[
                            (crash_parties_car_data['Vehicle1Model'] == 'PICK UP') & 
                            crash_parties_car_data['Vehicle1Year'].isna() & 
                            crash_parties_car_data['Vehicle1Make'].isna()].index, inplace = True)

In [230]:
crash_parties_car_data.drop(crash_parties_car_data[(crash_parties_car_data['Vehicle1Model'] == 'PICK UP') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'UNKNOWN')].index, inplace = True)

In [231]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'PICK UP') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'RAM'), 'Vehicle1Model'] = 'RAM 1500'

In [232]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'TRUCK') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'RAM'), 'Vehicle1Model'] = 'RAM 1500'

In [233]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'].isin (['PICK UP', 'PICKUP'])) &
                            (crash_parties_car_data['Vehicle1Make'] == 'NISSAN'), 'Vehicle1Model'] = 'FRONTIER'

In [234]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'TRUCK') &
                            (crash_parties_car_data['Vehicle1Make'] == 'NISSAN'), 'Vehicle1Model'] = 'FRONTIER'

In [235]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'PICK UP') &
                            (crash_parties_car_data['Vehicle1Make'] == 'CHEVROLET'), 'Vehicle1Model'] = 'SILVERADO'

In [236]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'TRUCK') &
                            (crash_parties_car_data['Vehicle1Make'] == 'CHEVROLET'), 'Vehicle1Model'] = 'SILVERADO'

In [237]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'PICK UP') &
                            (crash_parties_car_data['Vehicle1Make'] == 'FORD'), 'Vehicle1Model'] = 'F-150'

In [238]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'TRUCK') &
                            (crash_parties_car_data['Vehicle1Make'] == 'FORD'), 'Vehicle1Model'] = 'F-150'

In [239]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'PICK UP') &
                            (crash_parties_car_data['Vehicle1Make'] == 'DODGE'), 'Vehicle1Model'] = 'RAM 1500'

In [240]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'TRUCK') &
                            (crash_parties_car_data['Vehicle1Make'] == 'DODGE'), 'Vehicle1Model'] = 'RAM 1500'

In [241]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'TRUCK') &
                            (crash_parties_car_data['Vehicle1Make'] == 'JEEP'), 'Vehicle1Model'] = 'GLADIATOR'

In [242]:
crash_parties_car_data.drop(crash_parties_car_data[(crash_parties_car_data['Vehicle1Model'] == 'PICK UP') & (crash_parties_car_data['Vehicle1Make'] == 'MAZDA')].index,inplace = True)

In [243]:
crash_parties_car_data.drop(crash_parties_car_data[
                            (crash_parties_car_data['Vehicle1Model'] == 'UNKNOWN') ].index, inplace = True) # Would take too long to cross reference all 900 rows for what car they are based on just make

In [244]:
crash_parties_car_data.drop(crash_parties_car_data[
                            (crash_parties_car_data['Vehicle1Model'] == 'UNKNOWN') ].index, inplace = True) # Would take too long to cross reference all 900 rows for what car they are based on just make

In [245]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == '579') & (crash_parties_car_data['Vehicle1Make'] == 'SEMI-TRUCK'),'Vehicle1Model'] = 'SEMI-TRUCK'

In [246]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == '389') & (crash_parties_car_data['Vehicle1Make'] == 'SEMI-TRUCK'),'Vehicle1Model'] = 'SEMI-TRUCK'

In [247]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == '3 AXLE') & (crash_parties_car_data['Vehicle1Make'] == 'TRAILER'),'Vehicle1Model'] = 'TRAILER'

In [248]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == '3 AXLE') & (crash_parties_car_data['Vehicle1Make'] == 'SEMI-TRUCK'),'Vehicle1Model'] = 'SEMI-TRUCK'

In [249]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == '3 AXLE'), 'Vehicle1Model'] = 'SEMI-TRUCK' #Three wheel vehicles appear to be mainly semi trucks

In [250]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == '3AXLE'), 
'Vehicle1Model'] = 'SEMI-TRUCK' #Three wheel vehicles appear to be mainly semi trucks

In [251]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'TRUCK') & (crash_parties_car_data['Vehicle1Make'] == 'ROGU'), 'Vehicle1Model'] = 'DUMP TRUCK'

In [252]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'TRUCK') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'KME'), 'Vehicle1Model'] = 'Fire Truck'

In [253]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'TRUCK') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'PIERCE'), 'Vehicle1Model'] = 'Fire Truck'

In [254]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'TRUCK') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'FRIE'), 'Vehicle1Model'] = 'Fire Truck'

In [255]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'TRUCK') & (crash_parties_car_data['Vehicle1Make'] == 'SEMI-TRUCK'), 'Vehicle1Model'] = 'SEMI-TRUCK'

In [256]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'TRUCK') & (crash_parties_car_data['Vehicle1Make'] == 'VOLVO'), 'Vehicle1Model'] = 'SEMI-TRUCK'

In [257]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'TRUCK') & (crash_parties_car_data['Vehicle1Make'] == 'TESLA'), 'Vehicle1Model'] = 'CYBERTRUCK'

In [258]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'TRUCK') & (crash_parties_car_data['Vehicle1Make'] == 'ISUZU'), 'Vehicle1Model'] = 'BOX TRUCK'

In [259]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'TRUCK') & (crash_parties_car_data['Vehicle1Make'] == 'UTILITY'), 'Vehicle1Model'] = 'UTILITY VAN'

In [260]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'SUV') & 
                                                    (crash_parties_car_data['Vehicle1Make'] == 'RIVIAN'), 'Vehicle1Model'] = 'R1S'

In [261]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'PICKUP') & 
                                                    (crash_parties_car_data['Vehicle1Make'] == 'RIVIAN'), 'Vehicle1Model'] = 'R1T'

In [262]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'PICK UP') & 
                                                    (crash_parties_car_data['Vehicle1Make'] == 'RIVIAN'), 'Vehicle1Model'] = 'R1T'

In [263]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'PICK  UP') & 
                                                    (crash_parties_car_data['Vehicle1Make'] == 'RIVIAN'), 'Vehicle1Model'] = 'R1T'

In [264]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'TRUCK') & 
                                                    (crash_parties_car_data['Vehicle1Make'] == 'RIVIAN'), 'Vehicle1Model'] = 'R1T'

In [265]:
crash_parties_car_data.drop(crash_parties_car_data[(crash_parties_car_data['Vehicle1Make'] == 'RIVIAN') &
                              (crash_parties_car_data['Vehicle1Model'].isna())].index, inplace = True)

In [266]:
crash_parties_car_data.drop(crash_parties_car_data[(crash_parties_car_data['Vehicle1Make'] == 'RIVIAN') & 
                            ((crash_parties_car_data['Vehicle1Model'] != 'R1S') &  
                             (crash_parties_car_data['Vehicle1Model'] != 'R1T') & 
                             (crash_parties_car_data['Vehicle1Model'] != 'VAN'))].index, inplace = True)

In [267]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'TRUCK'), 'Vehicle1Model'] = 'SEMITRUCK'

In [268]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'ES')]

,CollisionId,Vehicle1Year,Vehicle1Make,Vehicle1Model,Vehicle1Color,V1IsVehicleTowed
3707,4537890,2022.0,LEXUS,ES,White,False
11928,4549249,2020.0,LEXUS,ES,BLK,False
15310,4546079,2009.0,LEXUS,ES,WHI,False
19287,4542796,2008.0,LEXUS,ES,BLACK,False
20136,4561789,2007.0,LEXUS,ES,BLU,False
...,...,...,...,...,...,...
749883,5037294,2014.0,LEXUS,ES,WHI,False
750959,5105117,2011.0,LEXUS,ES,WHITE,False
755972,5075819,2022.0,LEXUS,ES,WHI,False
763877,5115955,2018.0,LEXUS,ES,NaN,False


In [269]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == '2 AXLE') & 
                            ((crash_parties_car_data['Vehicle1Make'] == 'SEMI-TRUCK') | (crash_parties_car_data['Vehicle1Make'] == 'VOLVO')), 
                             'Vehicle1Model'] = 'SEMI-TRUCK'

In [270]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == '2 AXLE') & 
                            ((crash_parties_car_data['Vehicle1Make'] == 'TRAILER')), 
                             'Vehicle1Model'] = 'TRAILER'

In [271]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == '2 AXLE') & 
                            ((crash_parties_car_data['Vehicle1Make'] == 'TRAILER')), 
                             'Vehicle1Model'] = 'TRAILER'

In [272]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == '2 AXLE') & 
                            ((crash_parties_car_data['Vehicle1Make'] == 'UTILITY')), 
                             'Vehicle1Model'] = 'UTILITY VAN'

In [273]:
crash_parties_car_data.drop(crash_parties_car_data[(crash_parties_car_data['Vehicle1Model'] == 'SUV') & 
                            ((crash_parties_car_data['Vehicle1Make'].isna()) | 
                             (crash_parties_car_data['Vehicle1Make'] == 'UNKNOWN') |  
                            (crash_parties_car_data['Vehicle1Make'] == 'HIT & RUN VEHICLE'))].index, inplace = True)

In [274]:
crash_parties_car_data.drop(crash_parties_car_data[(crash_parties_car_data['Vehicle1Model'] == 'SUV') & 
                            crash_parties_car_data['Vehicle1Year'].isna()].index, inplace = True)

In [275]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'SUV') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'TOYOTA'), 'Vehicle1Model'] = 'RAV4'

In [276]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'SUV') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'TESLA') & 
                            (crash_parties_car_data['Vehicle1Year'] < 2020), 'Vehicle1Model'] = 'MODEL X'

In [277]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'SUV') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'HONDA'), 'Vehicle1Model'] = 'CRV'

In [278]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'SUV') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'FORD'), 'Vehicle1Model'] = 'EXPLORER'

In [279]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'SUV') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'NISSAN'), 'Vehicle1Model'] = 'ROGUE'

In [280]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'SUV') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'MAZDA'), 'Vehicle1Model'] = 'CX-5'

In [281]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'SUV') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'JEEP'), 'Vehicle1Model'] = 'WRANGLER'

In [282]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'SUV') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'ACURA'), 'Vehicle1Model'] = 'MDX'

In [283]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'SUV') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'CHEVROLET'), 'Vehicle1Model'] = 'EQUINOX'

In [284]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'SUV') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'BMW'), 'Vehicle1Model'] = 'X5'

In [285]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'SUV') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'LEXUS'), 'Vehicle1Model'] = 'RX350'

In [286]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'SUV') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'HYUNDAI'), 'Vehicle1Model'] = 'SANTA FE'

In [287]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'SUV') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'KIA'), 'Vehicle1Model'] = 'SORENTO'

In [288]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'SUV') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'AUDI'), 'Vehicle1Model'] = 'Q5'

In [289]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'SUV') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'MERCEDES-BENZ'), 'Vehicle1Model'] = 'GLC 300'

In [290]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'SUV') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'LAND ROVER'), 'Vehicle1Model'] = 'DEFENDER'

In [291]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'SUV') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'CADILLAC'), 'Vehicle1Model'] = 'ESCALADE'

In [292]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'SUV') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'SUBARU'), 'Vehicle1Model'] = 'CROSSTREK'

In [293]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'SUV') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'VOLKSWAGEN'), 'Vehicle1Model'] = 'TIGUAN'

In [294]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'SUV') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'BUICK'), 'Vehicle1Model'] = 'ENCORE'

In [295]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'SUV') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'INFINITI'), 'Vehicle1Model'] = 'QX60'

In [296]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'SUV') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'PORSCHE'), 'Vehicle1Model'] = 'CAYENNE'

In [297]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'SUV') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'MITSUBISHI'), 'Vehicle1Model'] = 'OUTLANDER'

In [298]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'SUV') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'VOLVO'), 'Vehicle1Model'] = 'XC90'

In [299]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'SUV') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'GMC'), 'Vehicle1Model'] = 'YUKON'

In [300]:
crash_parties_car_data.drop(crash_parties_car_data[(crash_parties_car_data['Vehicle1Model'] == 'SUV') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'LINCOLN')].index, inplace = True) #dropping Lincoln SUVs since there is not any existing records in the top 500

In [301]:
crash_parties_car_data.drop(crash_parties_car_data[(crash_parties_car_data['Vehicle1Model'] == 'SUV')].index, inplace = True) #MIFU doesnt make suvs

In [302]:
crash_parties_car_data.drop(crash_parties_car_data[(crash_parties_car_data['Vehicle1Model'] == '2 AXLE')].index, inplace = True) #dropping the 2 Axle records since its a small amount of rows, it is not important if they get dropped 

In [303]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'ES') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'LEXUS') & 
                            (crash_parties_car_data['Vehicle1Year'] < 2006), 'Vehicle1Model'] = 'ES300'

In [304]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'ES') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'LEXUS') & 
                            (crash_parties_car_data['Vehicle1Year'] >= 2006), 'Vehicle1Model'] = 'ES350'

In [305]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'CIVIC') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'TOYOTA'), 'Vehicle1Make'] = 'HONDA'

In [306]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'].isna()) & 
                            (crash_parties_car_data['Vehicle1Make'] == 'TOYOTA')&
                            (crash_parties_car_data['Vehicle1Year'] == 2006)]

,CollisionId,Vehicle1Year,Vehicle1Make,Vehicle1Model,Vehicle1Color,V1IsVehicleTowed
10522,4550524,2006.0,TOYOTA,NaN,NaN,False
10657,4550404,2006.0,TOYOTA,NaN,NaN,False
17605,4544170,2006.0,TOYOTA,NaN,NaN,False
17674,4544131,2006.0,TOYOTA,NaN,NaN,False
22120,4557751,2006.0,TOYOTA,NaN,NaN,False
...,...,...,...,...,...,...
756627,5072393,2006.0,TOYOTA,NaN,NaN,False
758318,5069580,2006.0,TOYOTA,NaN,NaN,False
759763,5065249,2006.0,TOYOTA,NaN,NaN,False
765529,5111649,2006.0,TOYOTA,NaN,NaN,False


In [307]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'CAMRY') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'TESLA')]

,CollisionId,Vehicle1Year,Vehicle1Make,Vehicle1Model,Vehicle1Color,V1IsVehicleTowed
223573,4689465,2022.0,TESLA,CAMRY,WHI,False


In [308]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'MODEL 3') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'TESLA') &
                            (crash_parties_car_data['Vehicle1Year'] < 2017), 'Vehicle1Model'] = 'MODEL S'

In [309]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'MODEL Y') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'TESLA') &
                            (crash_parties_car_data['Vehicle1Year'] < 2020), 'Vehicle1Model'] = 'MODEL X'

In [310]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'MODEL 3') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'TESLA') &
                            (crash_parties_car_data['Vehicle1Year'] < 2017)]

,CollisionId,Vehicle1Year,Vehicle1Make,Vehicle1Model,Vehicle1Color,V1IsVehicleTowed


In [311]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'MODEL Y') & 
                            (crash_parties_car_data['Vehicle1Make'] == 'TESLA') &
                            (crash_parties_car_data['Vehicle1Year'] < 2020)]

,CollisionId,Vehicle1Year,Vehicle1Make,Vehicle1Model,Vehicle1Color,V1IsVehicleTowed


In [312]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'PRIUS') & 
                            (crash_parties_car_data['Vehicle1Make'] != 'TOYOTA'), 'Vehicle1Make'] = 'TOYOTA'

In [313]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'CAMRY') & 
                            (crash_parties_car_data['Vehicle1Make'] != 'TOYOTA'), 'Vehicle1Make'] = 'TOYOTA'

In [314]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'COROLLA') & 
                            (crash_parties_car_data['Vehicle1Make'] != 'TOYOTA'), 'Vehicle1Make'] = 'TOYOTA'

In [315]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'RAV4') & 
                            (crash_parties_car_data['Vehicle1Make'] != 'TOYOTA'), 'Vehicle1Make'] = 'TOYOTA'

In [316]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'ACCORD') & 
                            (crash_parties_car_data['Vehicle1Make'] != 'HONDA'), 'Vehicle1Make'] = 'HONDA'

In [317]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'CIVIC') & 
                            (crash_parties_car_data['Vehicle1Make'] != 'HONDA'), 'Vehicle1Make'] = 'HONDA'

In [318]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'ACCORD') & 
                            (crash_parties_car_data['Vehicle1Make'] != 'HONDA'), 'Vehicle1Make'] = 'HONDA'

In [319]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'SILVERADO') & 
                            (crash_parties_car_data['Vehicle1Make'] != 'CHEVROLET'), 'Vehicle1Make'] = 'CHEVROLET'

In [320]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'].isin(['F-150','FUSION', 'C-MAX'])) & 
                            (crash_parties_car_data['Vehicle1Make'] != 'FORD'), 'Vehicle1Make'] = 'FORD'

In [321]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'CRV') & 
                            (crash_parties_car_data['Vehicle1Make'] != 'HONDA'), 'Vehicle1Make'] = 'HONDA'

In [322]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'].isin( ['ALTIMA','ROGUE'])) & 
                            (crash_parties_car_data['Vehicle1Make'] != 'NISSAN'), 'Vehicle1Make'] = 'NISSAN'

In [323]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'SIERRA') & 
                            (crash_parties_car_data['Vehicle1Make'] != 'GMC'), 'Vehicle1Make'] = 'GMC'

In [324]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'MODEL 3') & 
                            (crash_parties_car_data['Vehicle1Make'] != 'TESLA'), 'Vehicle1Make'] = 'TESLA'

In [325]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'MODEL Y') & 
                            (crash_parties_car_data['Vehicle1Make'] != 'TESLA'), 'Vehicle1Make'] = 'TESLA'

In [326]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'RAM 1500') & 
                            (crash_parties_car_data['Vehicle1Make'] != 'RAM'), 'Vehicle1Make'] = 'RAM'

In [327]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'].isin( ['MPV'])) & 
                            (crash_parties_car_data['Vehicle1Make'] != 'RAM'), 'Vehicle1Make'] = 'RAM'

In [328]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'ELANTRA') & 
                            (crash_parties_car_data['Vehicle1Make'] != 'HYUNDAI'), 'Vehicle1Make'] = 'HYUNDAI'

In [329]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'HIGHLANDER') & 
                            (crash_parties_car_data['Vehicle1Make'] != 'TOYOTA'), 'Vehicle1Make'] = 'TOYOTA'

In [330]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'TACOMA') & 
                            (crash_parties_car_data['Vehicle1Make'] != 'TOYOTA'), 'Vehicle1Make'] = 'TACOMA'

In [331]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'].isin(['PILOT','MDX','ODSSEY'])) & 
                            (crash_parties_car_data['Vehicle1Make'] != 'HONDA'), 'Vehicle1Make'] = 'HONDA'

In [332]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'].isin(['HRV','HR-V'])) & 
                            (crash_parties_car_data['Vehicle1Make'] != 'HONDA'), 'Vehicle1Make'] = 'HONDA'

In [333]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'] == 'BRZ') & 
                            (crash_parties_car_data['Vehicle1Make'] != 'SUBARU'), 'Vehicle1Make'] = 'SUBARU'

In [334]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'].isin(['EQUINOX','COLORADO'])) & 
                            (crash_parties_car_data['Vehicle1Make'] != 'CHEVROLET'), 'Vehicle1Make'] = 'CHEVROLET'

In [335]:
crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'].isin(['Q7','Q3'])) & 
                            (crash_parties_car_data['Vehicle1Make'] != 'AUDI'), 'Vehicle1Make'] = 'AUDI'

In [336]:
crash_parties_car_data.drop(crash_parties_car_data[(crash_parties_car_data['Vehicle1Model'] == 'ES')].index,inplace= True)

In [337]:
crash_parties_car_data.drop(crash_parties_car_data[crash_parties_car_data['Vehicle1Model'] =='TOYOTA'].index, inplace = True)

In [338]:
crash_parties_car_data.drop(crash_parties_car_data.loc[(crash_parties_car_data['Vehicle1Model'].isna())].index, inplace = True)

In [339]:
len(crash_parties_car_data)

570635

In [340]:
crash_parties_car_data.loc[crash_parties_car_data['Vehicle1Model'].isna()]

,CollisionId,Vehicle1Year,Vehicle1Make,Vehicle1Model,Vehicle1Color,V1IsVehicleTowed


In [341]:
crash_parties_car_data['Vehicle1Model'].value_counts()[0:50]

Vehicle1Model
CAMRY            24288
CIVIC            23829
ACCORD           22431
COROLLA          18707
SILVERADO        12960
TACOMA           11846
PRIUS            11546
RAV4              9726
ALTIMA            9188
TRAILER           9114
CRV               8618
MODEL 3           7613
F-150             7025
SENTRA            6757
MODEL Y           6341
RAM 1500          5872
ELANTRA           5456
EXPLORER          5413
SIENNA            5034
HIGHLANDER        4960
SIERRA            4592
CASCADIA          4541
ROGUE             4491
TUNDRA            4095
3 SERIES          4095
FUSION            4020
SONATA            3927
MUSTANG           3854
TRUCK TRACTOR     3824
CHARGER           3743
TAHOE             3737
MALIBU            3529
F-250             3520
PILOT             3506
4-RUNNER          3347
JETTA             3274
FORTE             3209
ODYSSEY           3095
ESCAPE            2967
WRANGLER          2890
FORD              2884
UTILITY VAN       2875
VERSA             27

In [342]:
crash_parties_car_data['Vehicle1Model'].value_counts()[50:100]

Vehicle1Model
CHEROKEE          2207
YUKON             2102
SEMI-TRUCK        2094
FORESTER          2063
F-350             1979
GRAND CHEROKEE    1952
HR-V              1904
TRANSIT           1899
SOUL              1894
COLORADO          1853
SUBURBAN          1853
CAMARO            1846
RX350             1842
SORENTO           1789
CX-5              1744
CRUZE             1725
BUS               1698
MDX               1691
TRAVERSE          1686
CHALLENGER        1671
DURANGO           1661
EDGE              1652
SANTA FE          1603
MAZDA3            1593
CROSSTREK         1547
AVALON            1545
EXPEDITION        1539
PATHFINDER        1537
FIT               1534
C300              1510
ES350             1484
TL                1434
IS250             1395
SPORTAGE          1387
ES300             1299
IMPALA            1246
YARIS             1242
RAM 3500          1241
RAM               1234
X5                1167
MODEL S           1163
300 Series        1134
X3                11

In [343]:
crash_parties_car_data['Vehicle1Model'].value_counts()[100:150]

Vehicle1Model
ACCENT         1014
MOTORCYCLE      996
TUSCON          992
A4              956
PASSAT          943
MODEL X         909
Q5              908
MAXIMA          907
OUTLANDER       904
Q50             878
UTILITY         878
RANGE ROVER     876
BRONCO          868
TERRAIN         852
NIRO            849
PACIFICA        847
TRAX            845
KICKS           842
MAZDA6          836
RDX             825
T680            821
TSX             811
K5              775
MURANO          771
SEQUOIA         771
F-550           771
TC              771
ACADIA          749
VOLT            745
TITAN           732
KONA            716
TUCSON          705
TAURUS          698
VENZA           698
C350            695
XB              685
G37             660
CTS             635
MAVERICK        630
JOURNEY         627
GLC 300         617
FIESTA          617
MATRIX          610
BLAZER          599
BOLT            593
MC              590
RIO             586
GOLF            586
F-450           580
CANYON

In [344]:
crash_parties_car_data['Vehicle1Model'].value_counts()[150:200]

Vehicle1Model
DAKOTA            577
WRX               576
ATLAS             572
TRAIL BLAZER      564
COOPER            550
LEGACY            535
FLATBED           534
INSIGHT           526
200               524
S10               520
CORVETTE          517
CX5               517
TELLURIDE         515
IS300             504
IS350             500
A3                500
FLEX              500
INTEGRA           495
SCION             492
TLX               492
M2                488
Q7                485
IONIQ             484
PATRIOT           475
CX-30             473
X1                470
RIDGELINE         470
DUMP TRUCK        450
XC90              449
RENEGADE          447
BICYCLE           438
AVALANCHE         437
LANCER            432
XTERRA            431
LIBERTY           431
SOLARA            424
BOX TRUCK         419
ENCORE            418
TOWN & COUNTRY    415
GENESIS           413
528I              412
SAVANNA           406
ILX               404
ARMADA            402
530I              

In [345]:
crash_parties_car_data['Vehicle1Model'].value_counts()[200:250]

Vehicle1Model
GRAND CARAVAN     369
K4                369
CROWN VICTORIA    368
GLADIATOR         361
ELEMENT           360
ENCLAVE           359
CAYENNE           358
NAVIGATOR         357
DART              354
PROMASTER         353
MIRAGE            348
PALISADE          346
ASCENT            346
C-MAX             346
VNL               337
SEDAN             336
A5                335
GS300             334
E-CLASS           330
Q3                329
NX300             329
PT CRUISER        327
GLE350            323
SONIC             323
IONIQ 5           317
A6                315
MINI COOPER       314
ECONOLINE E350    313
500               311
SPARK             306
350Z              303
MACAN             303
GS350             302
430I              295
S60               293
COBALT            292
I4                292
TRAILBLAZER       290
ID4               289
LEAF              287
C-HR              286
E320              283
NV200             281
MKZ               279
CT200H            

In [346]:
crash_parties_car_data['Vehicle1Model'].value_counts()[250:300]

Vehicle1Model
TAOS               266
CLARITY            266
ML350              263
CYBERTRUCK         263
SRX                262
SELTOS             261
ENVOY              257
SEDONA             256
LACROSSE           253
CHASSIS AND CAB    252
MIATA              251
EV6                250
PASSPORT           250
QUEST              249
NX350              247
CH-R               245
ACC                244
ES330              243
XC60               237
GL450              237
DENALI             234
CLA250             234
COROLLA CROSS      234
PROLOGUE           232
AVENGER            232
VUE                229
T880               229
CIV                228
RX                 228
CX-9               227
BOX                223
E300               223
RSX                215
CX-50              213
F650 SUPER CREW    212
ECONOLINE          211
DISCOVERY          210
RX330              210
FJ CRUISER         209
TANKER             209
4500               207
S550               207
TOWN CAR           2

In [347]:
crash_parties_car_data['Vehicle1Model'].value_counts()[300:350]

Vehicle1Model
ASTRO VAN          188
GRAND MARQUIS      186
BZ4X               183
X6                 183
XTS                182
ES250              182
G6                 182
IS                 182
CELICA             181
350                181
BRZ                181
ION                180
JUKE               180
GLA250             180
SCION TC           178
QX50               177
567                175
X7                 173
R6                 173
NPR                171
FX35               170
ROAD GLIDE         170
C230               169
ATS                167
300 SERIES         166
PICKUP             166
E450               166
TANK               166
STINGER            165
CLA 250            164
328                162
CALIBER            161
TRANSIT CONNECT    161
SCOOTER            158
LT                 157
428I               156
323                155
MV607              155
E-BIKE             153
OTHER              153
VOYAGER            152
VENUE              151
IX                 1

In [348]:
crash_parties_car_data['Vehicle1Model'].value_counts()[350:400]

Vehicle1Model
ARIYA            144
STREET GLIDE     143
NX               142
VIBE             142
AVEO             141
WAGONEER         140
X4               140
TOW TRUCK        139
MX5              139
540I             138
MIRAI            138
TT               138
Q8               136
Q60              134
FLAT BED         134
MKX              133
SONOMA           133
GX460            133
ECONOLINE 100    133
LS430            132
LS460            132
RX3OO            131
LS400            131
COUNTRYMAN       131
EXCURSION        131
150              129
NX200T           128
250 SERIES       127
ECOSPORT         127
268              127
CENTURY          127
XT5              126
METRIS           125
CROWN            125
GLA 250          125
F15              123
DEFENDER         123
FRS              123
GLE              123
XPL              122
CARRIER          121
NRR              120
XC40             120
XA               119
SEBRING          118
370Z             118
SLV              118

In [349]:
crash_parties_car_data['Vehicle1Model'].value_counts()[400:450]

Vehicle1Model
740I              117
IA                116
525I              116
A220              116
QX56              115
RX400             115
CROSSTOUR         114
2                 113
I3                113
5                 113
530E              112
S-CLASS           112
ROAD KING         112
XD                111
ASTRO             110
GX470             109
CUBE              108
REBEL             108
GV80              108
Q4                107
TRANSLEAD         107
SPORTSTER         106
MV                106
GLK350            106
300C              105
XT4               105
MAGNUM            104
CAVALIER          104
C240              104
G70               103
X2                102
VNL760            102
550I              102
T-100             101
PANAMERA          101
SUPRA             101
535               100
PROTEGE            99
MACH E             99
LT625              99
E250               98
GLC                98
330E               98
CC                 97
ECONOLINE E150    

In [350]:
crash_parties_car_data['Vehicle1Model'].value_counts()[450:500]

Vehicle1Model
GSXR                94
TK                  94
LE SABRE            94
MONTERO             94
ENVISION            94
PRELUDE             93
S500                93
440I                93
T3                  93
BTM                 93
325                 92
LS                  92
911                 92
ML320               92
LAREDO              91
R3                  91
EV9                 91
ECONOLINE E250      90
228I                90
RAPTOR              90
R1T                 90
ROUGE               89
XC70                87
TAYCAN              87
3X                  87
M35                 87
REEFER              87
5 SERIES            86
TRASH TRUCK         86
GRAND HIGHLANDER    86
CX-90               85
E-450               85
Z4                  85
PULL TRAILER        85
COOPER S            84
D13                 84
GHIBLI              84
ENVISTA             84
M2 106              84
LATITUDE            83
MOUNTAINEER         83
R1                  83
TOUAREG             

After cleaning the vehicle 1 make so that there is less typos and data entry errors in the dataset, it seems to show that there needs to be better standards so less room for errors can be made

Making Category Class Labels for Easier Visualization 

In [351]:
crash_parties_car_data.dropna(subset=['Vehicle1Year'],inplace = True)

In [352]:
crash_parties_car_data.drop(crash_parties_car_data[crash_parties_car_data['Vehicle1Year'] < 1980].index,inplace = True)

In [353]:
crash_parties_car_data['Vehicle1Year'].isna()

1         False
2         False
3         False
4         False
5         False
          ...  
765984    False
766193    False
766407    False
766490    False
766827    False
Name: Vehicle1Year, Length: 561294, dtype: bool

In [354]:
crash_parties_car_data.to_csv("crash_parties_car_data.csv")

In [355]:
crash_parties_background_info = crash_parties_data[['CollisionId','PartyType','Special Information','IsAtFault']]

In [356]:
crash_parties_background_info

,CollisionId,PartyType,Special Information,IsAtFault
0,4541902,Driver,CELL PHONE USE UNKNOWN,True
1,4541901,Driver,CELL PHONE NOT IN USE,NaN
2,4541901,Driver,CELL PHONE USE UNKNOWN,True
3,4541900,Driver,CELL PHONE NOT IN USE,NaN
4,4541900,Driver,CELL PHONE NOT IN USE,NaN
...,...,...,...,...
767047,5105674,Driver,CELL PHONE NOT IN USE,True
767048,5105673,Driver,CELL PHONE NOT IN USE,True
767049,5105672,Driver,CELL PHONE NOT IN USE,NaN
767050,5105672,Driver,CELL PHONE NOT IN USE,True


In [357]:
crash_parties_background_info.loc[crash_parties_background_info['IsAtFault'].isna(), 'IsAtFault'] =  False

In [358]:
crash_parties_background_info

,CollisionId,PartyType,Special Information,IsAtFault
0,4541902,Driver,CELL PHONE USE UNKNOWN,True
1,4541901,Driver,CELL PHONE NOT IN USE,False
2,4541901,Driver,CELL PHONE USE UNKNOWN,True
3,4541900,Driver,CELL PHONE NOT IN USE,False
4,4541900,Driver,CELL PHONE NOT IN USE,False
...,...,...,...,...
767047,5105674,Driver,CELL PHONE NOT IN USE,True
767048,5105673,Driver,CELL PHONE NOT IN USE,True
767049,5105672,Driver,CELL PHONE NOT IN USE,False
767050,5105672,Driver,CELL PHONE NOT IN USE,True


In [359]:
crash_parties_background_info.to_csv("crash_parties_background_info.csv")

In [360]:
crash_parties_sobriety_data = crash_parties_data[['CollisionId','SobrietyDrugPhysicalCode1','SobrietyDrugPhysicalDescription1']]

In [361]:
crash_parties_data[['SobrietyDrugPhysicalCode1']].value_counts()

SobrietyDrugPhysicalCode1
A                            557347
G                             85662
H                             57382
B                             33966
D                              8755
C                              5110
E                              2268
I                              1470
F                               496
Name: count, dtype: int64

In [362]:
crash_parties_data[['SobrietyDrugPhysicalCode2']].value_counts()

SobrietyDrugPhysicalCode2
I                            3586
A                             851
E                             753
F                             443
B                             148
H                             143
G                              69
D                              15
C                               7
Name: count, dtype: int64

Since SobrietyDrugPhysicalCode2 is a very small fraction of the dataset, it is safe to not include this in the analysis going forward

In [363]:
crash_parties_data['SpeedLimit'].value_counts().head(10)

SpeedLimit
65.0    211033
55.0     78802
25.0     71706
35.0     63725
45.0     43642
40.0     43167
30.0     16560
50.0     16534
70.0     14009
15.0      5830
Name: count, dtype: int64

In [364]:
crash_parties_speed_limit_data = crash_parties_data[['CollisionId', 'SpeedLimit']]

In [365]:
crash_parties_data['Lane'].value_counts().head(10)

Lane
1      150737
2       90417
3       49871
4       34362
E/B     21847
S/B     21759
N/B     21573
W/B     21328
5       14153
HOV      7096
Name: count, dtype: int64

It seems that lane data might not be too helpful to the analysis since there is not enough information to understand this as I do not know how to handle direction for lane placements

In [366]:
crash_parties_speed_limit_data[crash_parties_speed_limit_data['CollisionId'] == 5105744]

,CollisionId,SpeedLimit
766923,5105744,25.0
766924,5105744,25.0
766925,5105744,25.0


In [367]:
crash_parties_data[crash_parties_data['CollisionId'] == 5105744]

,_id,PartyId,CollisionId,PartyNumber,PartyType,IsAtFault,IsOnDutyEmergencyVehicle,IsHitAndRun,AirbagCode,AirbagDescription,...,Vehicle2TypeDesc,Vehicle2Year,Vehicle2Make,Vehicle2Model,Vehicle2Color,V2IsVehicleTowed,Lane,ThruLanes,TotalLanes,IsDREConducted
766923,766924,14979745,5105744,3,ParkedVehicle,NaN,False,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,1,1.0,1.0,NaN
766924,766925,14979744,5105744,2,ParkedVehicle,NaN,False,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,1,1.0,1.0,NaN
766925,766926,14979743,5105744,1,Driver,True,False,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,1,1.0,1.0,NaN


In [368]:
crash_parties_speed_limit_data.to_csv('crash_parties_speed_limit_data.csv')